### DISCUSSION TITLE
"Fit::Crop" malfunction

### DISCUSSION THREAD

**Discussion by wemadefoxnever** [spatie/image#223] [Q&A] [answered]
Hi, everybody! I have a problem with Fit:crop function. Advice or help, please.

https://spatie.be/docs/image/v3/image-manipulations/resizing-images#content-fitcrop
**About Fit:Crop**
> Resizes the image to fill the width and height boundaries and crops any excess image data. The resulting image will match the width and height constraints without distorting the image.

My model config:

```
class Course extends Model implements HasMedia
{
    use InteractsWithMedia;
    public function registerMediaConversions(Media $media = null): void
    {
        $this
            ->addMediaConversion('cover')
            ->format('webp')
            ->fit(Fit::Crop, 600, 450)
            ->withResponsiveImages()
            ->optimize();
    }
}
```
As it appears, the script didn't crop the excess but merely compressed the image to the required dimensions. This strongly resembles a bug.
Original image:
![healthy-back](https://github.com/spatie/image/assets/1191825/5264d7d4-e2e1-4c83-a090-9a1d733a21bf)

Convertion image (Sorry for the JPG. GitHub doesn't support the WebP format, so I had to replace it with JPG.")
![ezgif-4-dc048d43d4](https://github.com/spatie/image/assets/1191825/05905128-44ac-48d3-b33f-f43b07fa1efe)

**Comment by whousehomestay** [comment]
Yeah, i got this too...

**Comment by lukasleitsch** [comment]
Having the same issue. I can reproduce the bug with this snippet.

```php
\Spatie\Image\Image::useImageDriver(ImageDriver::Gd)
    ->loadFile('example.jpg')
    ->fit(\Spatie\Image\Enums\Fit::Crop, 300, 300)
    ->save('example-cropped.jpg');
```

**Comment by dev** [comment]
Same issue here! v3.3.2

**Comment by timvandijck** [comment]
I just tagged a new version (3.3.3) that should resolve this issue.

**Reply by timvandijck** [reply]
Seems like there's still a small issue when upscaling, looking into it.

**Reply by ghost** [reply]
Hi. How do I upgrade this from Media Library Package? The media-library package still uses 3.3.2 and the issue still persist. Do I install this package to make it work?

**Reply by timvandijck** [reply]
@anuzpandey5 I believe you could just require 3.3.3 in your project composer file and media library should also use that one. I'll update the dependency in medialibrary when this feature is more stable.

**Reply by ghost** [reply]
@timvandijck Yes I did upgrade it and the problem is solved. Now the major issue is `Fit::Crop` has the behavior of `Fit::Fill`

**Comment by goyne** [comment]
I was facing the same issue using 3.3.2. After upgrading to 3.3.3 the distortion is gone, but white space is added to meet the given width and height boundaries (instead of cropping). Looks like Fit::Crop now has the behaviour of Fit::Fill.

![01HKJR8Z7GG47H57YWMA1BS18V-thumb](https://github.com/spatie/image/assets/2893526/47fcfe51-f7ab-4d7d-a1ee-db52bbf672c2)

Edited: not clearly visible due to white on white, but the photo has a white bar at the top and bottom. The original does not have that.

**Reply by whousehomestay** [reply]
Maybe because of this?

https://github.com/spatie/image/discussions/223#discussioncomment-8024622

**Reply by goyne** [reply]
Not sure if this is caused by upscaling. Original image is much larger.

**Reply by timvandijck** [reply]
Yeah, there are still some kinks to work out. I'm going to do some more work on it this week to improve things.

**Comment by timvandijck** [comment, marked as answer]
I just published versions 3.3.4 with some additional fixes to fit cropping. These should resolve the white borders you might get on certain images.
